logstash-core/src/main/java/org/logstash/ackedqueue/ext/JRubyAckedQueueExt.java

```diff
@@ -34,26 +34,39 @@ public Queue getQueue() {
         return this.queue;
     }
 
+    public static JRubyAckedQueueExt create(String path, int capacity, int maxEvents, int checkpointMaxWrites, int checkpointMaxAcks, long maxBytes) {
+        JRubyAckedQueueExt queueExt = new JRubyAckedQueueExt(RubyUtil.RUBY, RubyUtil.ACKED_QUEUE_CLASS);
+        queueExt.initializeQueue(path, capacity, maxEvents, checkpointMaxWrites, checkpointMaxAcks, maxBytes);
+        return queueExt;
+    }
+
     @JRubyMethod(name = "initialize", optional = 7)
     public IRubyObject ruby_initialize(ThreadContext context, IRubyObject[] args) {
         args = Arity.scanArgs(context.runtime, args, 7, 0);
         int capacity = RubyFixnum.num2int(args[1]);
         int maxUnread = RubyFixnum.num2int(args[2]);
+
+        // are these out of order?
```

Comment:
I believe `checkpointMaxAcks` and `checkpointMaxWrites` are being pulled from the wrong indexes in the argument array.